functions/supplier/sftp-supplier/src/test/java/org/springframework/cloud/fn/supplier/sftp/SftpSupplierApplicationTests.java

```diff
@@ -235,6 +240,46 @@ void deleteRemoteFiles() {
 
 	}
 
+	@Test
+	void renameRemoteFilesStream() {
+		defaultApplicationContextRunner
+				.withPropertyValues(
+						"sftp.supplier.stream=true",
+						"sftp.supplier.delete-remote-files=false",
+						"sftp.supplier.rename-remote-files-to='/sftpTarget/' + headers.get('file_remoteFile')")
```

Comment:
It can be simpler: `headers.file_remoteFile`. There is a special `MapPropertyAccessor` to treat the dot operator and the name after it as a `Map` key to call its `get()`.